src/Timeline/NewPost.js

```diff
@@ -0,0 +1,54 @@
+import { NewPostFrame, NewPostForm, NewPostURL, NewPostComment, PostLeftPanel } from "./NewPost_style.js";
+import { UserAvatar } from "./Timeline_style.js";
+import { useState, useContext } from "react";
+import { mandarPost } from '../services/api/Api';
+import { ContextPost } from "../services/contexts/ContextPost.js";
+
+export default function NewPost() {
+    const [urlLink, setUrlLink] = useState("");
+    const [texto, setTexto] = useState("");
+    const [status, setStatus] = useState("Publicar");
+    const [cor, setCor] = useState("");
+    const [disable, setDisable] = useState("");
+    const {userPostsArray, setUserPostsArray} = useContext(ContextPost);
+    
+    
+    const config = {
+        headers:{
+            Authorization: `Bearer 1b939c76-c0bc-48ae-979c-d1f020d74be6`
+        }
+    };
+
+    function Postagem(event){
+        event.preventDefault();
+        const body = {
+            text: texto,
+            link: urlLink
+        };
+        if(urlLink === "") return alert("preencha o link");
+        
+        if(status === "Publicar"){
+            setStatus("Publicando...");
+            setCor("desabilitar");
+            setDisable("unable");
```

Comment:
Poderíamos combinar esses 3 states em 1 só para evitar renderizar 3 vezes conforme for atualizar.